workers/python/extraction.py

```diff
@@ -0,0 +1,69 @@
+import asyncio
+
+from bullmq import Queue
+from unstructured_ingest.v2.pipeline.pipeline import Pipeline
+from unstructured_ingest.v2.interfaces import ProcessorConfig
+from unstructured_ingest.v2.processes.partitioner import PartitionerConfig
+from unstructured_ingest.v2.processes.connectors.fsspec.s3 import (
+    S3IndexerConfig, S3DownloaderConfig, S3ConnectionConfig, S3AccessConfig)
+from unstructured_ingest.v2.processes.connectors.fsspec.s3 import (
+    S3ConnectionConfig, S3AccessConfig, S3UploaderConfig)
+from unstructured_ingest.v2.processes.chunker import ChunkerConfig
+
+from workers import create_worker
+from config import EXTRACTION_BACKEND, REDIS_URL, S3_ENDPOINT, S3_BUCKET_FILE_STORAGE, S3_ACCESS_KEY_ID, S3_SECRET_ACCESS_KEY, UNSTRUCTURED_API_KEY, UNSTRUCTURED_API_URL
+from enums import ExtractionBackend
+
+EXTRACTION_QUEUE_NAME = "files:extraction"
+EXTRACTION_POSTPROCESSOR_QUEUE_NAME = "files:extraction:postprocess"
+
+EXTRACTION_DIR = "extractions"
+
+S3_URL = f"s3://{S3_BUCKET_FILE_STORAGE}"
+
+
+async def processExtraction(job, job_token):
+    data = job.data
+    storageId = data['storageId']
+    s3_connection_config = S3ConnectionConfig(
+        endpoint_url=S3_ENDPOINT,
+        access_config=S3AccessConfig(
+            key=S3_ACCESS_KEY_ID,
+            secret=S3_SECRET_ACCESS_KEY
+        )
+    )
+    pipeline = Pipeline.from_configs(
+        context=ProcessorConfig(),
+        indexer_config=S3IndexerConfig(remote_url=f"{S3_URL}/{storageId}"),
+        downloader_config=S3DownloaderConfig(),
+        source_connection_config=s3_connection_config,
+        partitioner_config=PartitionerConfig(
+            partition_by_api=EXTRACTION_BACKEND == ExtractionBackend.UNSTRUCTURED_API,
+            partition_endpoint=UNSTRUCTURED_API_URL,
+            api_key=UNSTRUCTURED_API_KEY
+        ),
+        chunker_config=ChunkerConfig(
+            chunk_by_api=EXTRACTION_BACKEND == ExtractionBackend.UNSTRUCTURED_API,
+            chunking_endpoint=UNSTRUCTURED_API_URL,
+            chunk_api_key=UNSTRUCTURED_API_KEY,
+            chunking_strategy="by_title",
+        ),
+        destination_connection_config=s3_connection_config,
+        uploader_config=S3UploaderConfig(
+            remote_url=f"{S3_URL}/{EXTRACTION_DIR}")
+    )
+    await asyncio.to_thread(pipeline.run)
```

Comment:
Note: probably ok, but we may want to specify maximum number of threads, the default is a bit unpredictable. (`asyncio.run_in_executor` with a custom threadpool executor)

Also this might break if worker is processing multiple tasks concurrently and `pipeline.run` is not threadsafe. Then we'll need to serialize it using executor with as single thread or `asgiref.sync_to_async`.

Reply:
> Also this might break if worker is processing multiple tasks concurrently and pipeline.run is not threadsafe.

Shouldn't matter to us due to GIL. We're using it to make a blocking call asynchronous.